### PR TITLE
Avoid expensive sqrt operation in hot loop of `BitMap.grow_mask`

### PR DESCRIPTION
**Pull request by LuizZak**
I've noticed that `BitMap.grow_mask` could be quite slow for large radii on large images, and while looking for ways to optimize the process I noticed that the operation performs a `distance_to` that could be converted into a `distance_squared_to`, avoiding a `sqrt` from being performed on every pixel, multiple times.

Guidance is requested on perhaps how to ensure this doesn't lead to an unwanted regression on pixels on the edge of expanded bitmaps.

### CODE REVIEW

File: scene/resources/bit_map.cpp [hunk 1]
```diff
@@ -588,8 +589,8 @@ void BitMap::grow_mask(int p_pixels, const Rect2i &p_rect) {
 						}
 					}
 
-					float d = Point2(j, i).distance_to(Point2(x, y)) - CMP_EPSILON;
-					if (d > p_pixels) {
+					float d = Point2(j, i).distance_squared_to(Point2(x, y)) - CMP_EPSILON;
```
Comment (akien-mga) [inline scene/resources/bit_map.cpp:592]:
```suggestion
					float d = Point2(j, i).distance_squared_to(Point2(x, y)) - CMP_EPSILON2;
```

File: scene/resources/bit_map.cpp [hunk 2]
```diff
@@ -559,6 +559,7 @@ void BitMap::grow_mask(int p_pixels, const Rect2i &p_rect) {
 
 	bool bit_value = p_pixels > 0;
 	p_pixels = Math::abs(p_pixels);
+	int p_pixels2 = p_pixels * p_pixels;
```
Comment (akien-mga) [inline scene/resources/bit_map.cpp:562]:
```suggestion
	const int pixels2 = p_pixels * p_pixels;
```

File: scene/resources/bit_map.cpp [hunk 3]
```diff
@@ -588,8 +589,8 @@ void BitMap::grow_mask(int p_pixels, const Rect2i &p_rect) {
 						}
 					}
 
-					float d = Point2(j, i).distance_to(Point2(x, y)) - CMP_EPSILON;
-					if (d > p_pixels) {
+					float d = Point2(j, i).distance_squared_to(Point2(x, y)) - CMP_EPSILON;
+					if (d > p_pixels2) {
```
Comment (akien-mga) [inline scene/resources/bit_map.cpp:593]:
```suggestion
					if (d > pixels2) {
```

**Comment by LuizZak** [comment]
I may have come up with a better solution involving an intermediary `BitMap` that holds pre-computed distances, but since this change is way simpler I'll opt to open a separate PR after this one gets closed.

**Review by AThousandShips — APPROVED** [review]
LGTM

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**